File: apps/typescript/test/fizzbuzz/domain/model/fizz-buzz-value.test.ts
import { describe, expect, test } from "vitest";
import { FizzBuzzValue } from "../../../../src/fizzbuzz/index.js";

describe("FizzBuzzValue", () => {
  test("値と数値を保持する", () => {
    const value = new FizzBuzzValue("Fizz", 3);

    expect(value.value).toBe("Fizz");
    expect(value.number).toBe(3);
  });

  test("toString は値を返す", () => {
    const value = new FizzBuzzValue("Buzz", 5);

    expect(value.toString()).toBe("Buzz");
  });

  test("同じ値と数値の場合 equals は true", () => {
    const v1 = new FizzBuzzValue("Fizz", 3);
    const v2 = new FizzBuzzValue("Fizz", 3);

    expect(v1.equals(v2)).toBe(true);
  });

  test("異なる値の場合 equals は false", () => {
    const v1 = new FizzBuzzValue("Fizz", 3);
    const v2 = new FizzBuzzValue("Buzz", 5);

    expect(v1.equals(v2)).toBe(false);
  });
});

File: apps/kotlin/src/test/kotlin/fizzbuzz/domain/model/FizzBuzzValueTest.kt
package fizzbuzz.domain.model

import fizzbuzz.domain.type.FizzBuzzType
import kotlin.test.Test
import kotlin.test.assertEquals

class FizzBuzzValueTest {
    @Test fun `値オブジェクトは数値を保持する`() =
        assertEquals(3, FizzBuzzValue.create(FizzBuzzType.TYPE_01, 3).number)

    @Test fun `値オブジェクトは変換結果を保持する`() =
        assertEquals("Fizz", FizzBuzzValue.create(FizzBuzzType.TYPE_01, 3).value)

    @Test fun `data class は等値比較できる`() =
        assertEquals(FizzBuzzValue(3, "Fizz"), FizzBuzzValue.create(FizzBuzzType.TYPE_01, 3))

    @Test fun `100 件のコレクションを生成する`() =
        assertEquals(100, FizzBuzzList.create(100, FizzBuzzType.TYPE_01).count)

    @Test fun `コレクションの 15 番目は FizzBuzz`() =
        assertEquals("FizzBuzz", FizzBuzzList.create(100, FizzBuzzType.TYPE_01).values[14].value)
}
